File: IRSClientAngular/src/app/component/generic-element/generic-element.ts
import { Config } from '../../util/config.service';
import { IncidentElement } from '../report/incident-element';

export class GenericElement implements IncidentElement {
    table: string;
    attributes: ElementAttributes;

    constructor() {
        this.attributes = new ElementAttributes();
        this.table = Config.GenericElementTable;
    }
}

export class ElementAttributes {
    GENERIC_ELEMENT_ID: number;
    TYPE: string;
    DESCRIPTION: string;
}

File: WebApiSpark/src/main/java/Model/GenericElement.java
package Model;


import Util.DatabaseValues;

public class GenericElement extends IncidentElement
{
    public GenericElement ( )
    {
        this (
                "",
                "",
                ""

        );
    }

    public GenericElement (
            String elementID,
            String type,
            String description
    ) {
        super (
                DatabaseValues.Table.GENERIC_ELEMENT,
                new DatabaseValues.Column[]
                        {
                                DatabaseValues.Column.GENERIC_ELEMENT_ID,
                                DatabaseValues.Column.TYPE,
                                DatabaseValues.Column.DESCRIPTION,
                        }
        );

        updateAttributeValue(
                DatabaseValues.Column.GENERIC_ELEMENT_ID,
                elementID
        );

        updateAttributeValue(
                DatabaseValues.Column.TYPE,
                type
        );

        updateAttributeValue(
                DatabaseValues.Column.DESCRIPTION,
                description
        );

    }
}
